flexget/plugins/input/gazelle.py

```diff
@@ -0,0 +1,499 @@
+        opts = self._opts(key)
+        if isinstance(opts, dict):
+            # Translate the input value to the API value
+            # The str cast converts bools to 'True'/'False' for use as keys
+            # This allows for options that have True/False/Other values
+            return opts[str(val)]
+        elif isinstance(val, list):
+            # Fix yaml parser making a list out of a string
+            return ",".join(val)
+        return val
+
+    def params_from_config(self, config):
+        """Filter params and map config values -> api values"""
+        ret = {}
+        for k, v in config.items():
+            key = self._key(k)
+            if key in self.params:
+                ret[key] = self._getval(k, v)
+        return ret
+
+    def setup(self, task, config):
+        """Set up a session and log in"""
+        self._session = task.requests
+        base_url = config.get('base_url', "").rstrip("/")
+        if base_url:
+            if self.base_url and self.base_url != base_url:
+                log.warning("Using plugin designed for %s on %s - "
+                            "things may break", self.base_url, base_url)
+            self.base_url = base_url
+
+        if not self.base_url:
+            raise PluginError("No 'base_url' configured")
+
+        # Any more than 5 pages is probably way too loose of a search
+        self.max_pages = config.get('max_pages', 5)
+
+        # The consistent request limiting rule seems to be:
+        # "Refrain from making more than five (5) requests every ten (10) seconds"
+        self._session.add_domain_limiter(TokenBucketLimiter(self.base_url, 2, '2 seconds'))
+
+        # Custom user agent
+        user_agent = config.get('user_agent', "Flexget [Gazelle plugin]")
+        self._session.headers.update({"User-Agent": user_agent})
+        self.username = config['username']
+        self.password = config['password']
+
+        # Login
+        self.authenticate()
+
+        # Logged in successfully, it's ok if nothing matches now
+        task.no_entries_ok = True
+
+    def resume_session(self):
+        """Resume an existing session from the datebase
+
+        Return True on successful recovery, False otherwise
+        """
+        log.debug("Attempting to find an existing session in the DB")
+        with Session() as session:
+            db_session = session.query(GazelleSession).filter(
+                GazelleSession.base_url == self.base_url,
+                GazelleSession.username == self.username
+            ).one_or_none()
+            if db_session and db_session.expires and db_session.expires >= datetime.utcnow():
+                # Found a valid session in the DB - use it
+                self._session.cookies.update(db_session.cookies)
+                self.authkey = db_session.authkey
+                self.passkey = db_session.passkey
+                return True
+        return False
```

Comment:
You don't _need_ to return False. No return statement is the same as `return None`

Reply:
The reason I explicitly return `False` is that having it return `True`/`False` vs. `True`/`None` is a little more logical IMO. Same reason why I don't return `True`/`0` even though it would behave the same way when used in truthyness comparisons.